react/InputPassword/index.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import cx from 'classnames'
+import { default as LabelStyles } from '../Label/styles.styl'
+import Input from '../Input'
+import styles from './styles.styl'
+
+class InputPassword extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { visible: false }
+    this.askShowHide = props.askShowHide !== false
+  }
+  toggleVisibility() {
+    this.setState({ visible: !this.state.visible })
+  }
+  render() {
+    return (
+      <div className={styles['wrapInputPassword']}>
+        {this.askShowHide && (
+          <div
+            className={cx(LabelStyles['c-label'], styles['actionOnInput'])}
+            onClick={() => this.toggleVisibility()}
+          >
+            {this.state.visible ? 'HIDE' : 'SHOW'}
+          </div>
+        )}
+        <Input
+          {...this.props}
+          type={this.state.visible ? 'text' : 'password'}
```

Comment:
As `this.state.visible` is used two times in this methond, maybe we can retrieve it at the beginning of the method with 
```js
const { visible } = this.state
```